gamemodes/terrortown/gamemode/client/cl_keys.lua

```diff
@@ -76,29 +77,38 @@ function GM:PlayerBindPress(ply, bindName, pressed)
             return TBHUD:UseFocused()
         end
 
-        local tr = util.TraceLine({
-            start = ply:GetShootPos(),
-            endpos = ply:GetShootPos() + ply:GetAimVector() * 100,
-            filter = ply,
-            mask = MASK_SHOT,
-        })
-
-        if not tr.Hit or not IsValid(tr.Entity) then
-            return
-        end
-
+        -- Find out if a marker is focussed otherwise check normal use
         local isClientOnly = false
-        if isfunction(tr.Entity.ClientUse) then
-            isClientOnly = tr.Entity:ClientUse()
+        local useEnt = markerVision.GetFocussedEntity()
+        local isRemote = IsValid(useEnt)
+        if not isRemote then
+            local tr = util.TraceLine({
+                start = ply:GetShootPos(),
+                endpos = ply:GetShootPos() + ply:GetAimVector() * 100,
+                filter = ply,
+                mask = MASK_SHOT,
+            })
+
+            useEnt = tr.Entity
+            if not tr.Hit or not IsValid(useEnt) then
+                return
+            end
+
+            if isfunction(useEnt.ClientUse) then
+                isClientOnly = useEnt:ClientUse()
+            end
+        elseif isfunction(useEnt.RemoteUse) then
+            isClientOnly = useEnt:RemoteUse(ply)
```

Comment:
Is there a reason `ply` is only passed to one of these two? If not, couldn't these two be combined?

Reply:
Well `RemoteUse` shall be a shared implementation, which was not possible with `Use` as it could have broken old implementations. Therefore ClientUse didnt need it, but RemoteUse does.

Reply:
I think I forgot how to read, to me it seemed like you call the same two functions here. My bad!